### PR TITLE
Dash 1.0 - Update version to 1.0.0a1

### PR DESCRIPTION
**Pull request by valentijnnieman**
This PR upgrades React to 16.6.1, and bumps the version up to `1.0.0-rc1`.

Relies on:
[dash-core-components 1.0.0](https://github.com/plotly/dash-core-components/pull/373)
[dash-html-components 1.0.0](https://github.com/plotly/dash-html-components/pull/81)
[dash 1.0.0](https://github.com/plotly/dash/pull/457)

===========================
@Marc-Andre-Rivet  is hijacking this existing PR from @valentijnnieman 😄 

Fixes https://github.com/plotly/dash-core/issues/4

* Bumps version to 1.0.0a1 (1.0.0-alpha1) instead of rc1 to better respect semver
* Updating internal references to 16.6.3 (bumping 16.2.0, dropping 15.x)
* Updating internal copy of react and react-dom min files
* Updated destination branch to `release-v1`
* Not taking care of dev-requirements.txt yet

### CODE REVIEW

**Comment by tcbegley** [comment]
Hey @nite, you can install with `pip install dash-renderer==1.0.0rc1`.

I'm not in a position to comment on the plans for merging, but since this introduces some pretty major changes I think it won't happen imminently. Check [this issue](https://github.com/plotly/dash/issues/469) for some pretty extensive discussion of the upcoming changes.

**Comment by Marc-Andre-Rivet** [comment]
@nite @tcbegley Thanks for your patience. The dash-renderer and dash-core-componentshave both previously been pre-released under 1.0.0rc1, this release will be deleted and replaced with 1.0.0a1. We are doing this in an attempt to better match semantic versioning expectations -- an `rc` release should have a fairly stable implementation and contract with the outside and this react version bump is but the first part of this Dash 1.0 transition, meaning that at this point in time there is no guarantee that the implementation is stable project wide or that we will not significantly change the contract.

As 1.0.0a1 < 1.0.0rc1 this may cause some inconvenience and I apologize for that.

File: src/components/core/Reloader.react.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ class Reloader extends React.Component {
                 ) {
                     // Look if it was a css file.
                     let was_css = false;
-                    for (let a of reloadRequest.content.files) {
+                    for (const a of reloadRequest.content.files) {
```
Comment (Marc-Andre-Rivet) [inline src/components/core/Reloader.react.js]: lint auto fix

File: tests/test_render.py [hunk 2]
```diff
@@ -1754,29 +1754,12 @@ def test_update_react_version(self):
             dash_renderer._js_dist_dependencies,
             [{
                 'external_url': [
-                    'https://unpkg.com/react@15.4.2/dist/react.min.js',
-                    'https://unpkg.com/react-dom@15.4.2/dist/react-dom.min.js',
+                    'https://unpkg.com/react@16.6.3/umd/react.production.min.js',
```
Comment (Marc-Andre-Rivet) [inline tests/test_render.py]: update our local and remote react versions

File: dash_renderer/__init__.py [hunk 3]
```diff
@@ -12,27 +12,17 @@
 from .version import __version__
 __file__
 
-_DEFAULT_REACT_VERSION = '15.4.2'
-_REACT_VERSION_TYPES = {'15.4.2', '16.2.0'}
+_DEFAULT_REACT_VERSION = '16.6.3'
+_REACT_VERSION_TYPES = {'16.6.3'}
 _REACT_VERSION_TO_URLS = {
-    '15.4.2': {
+    '16.6.3': {
```
Comment (Marc-Andre-Rivet) [inline dash_renderer/__init__.py]: Bump to 16.x, remove 15.x

File: dev-requirements.txt [hunk 4]
```diff
@@ -1,6 +1,6 @@
 dash_core_components==0.33.0
-dash_html_components==0.11.0rc5
-dash==0.32.0
+dash_html_components==0.13.4
```
Comment (Marc-Andre-Rivet) [inline dev-requirements.txt]: There's a fix in html comps for react 16, needs to be present for all tests to pass

File: dev-requirements.txt [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
+dash==0.32.2
```
Comment (bpostlethwaite) [inline dev-requirements.txt]: should we lock on to rc releases (github branch) here so we are testing against the latest of everything?

Comment (Marc-Andre-Rivet) [reply dev-requirements.txt]: We're already planning on making these point to head of master / release-v1 soon

**Comment by Marc-Andre-Rivet** [comment]
@bpostlethwaite A follow up to these is having a release-v1 of the dash-docs. Haven't included a task to test in at least of subset of existing apps when getting ready to push back into master. If you have a good subset in mind, would you mind creating an issue with some ideas on this in the `dash-core` repo and tagging it with `Dash 1.0` label?

**Pull request merged**